### ISSUE TITLE
Cannot save file opened by telescope

### ISSUE THREAD

**Issue by m-demare** [nvim-telescope/telescope.nvim#2965]
### Description

When opening some files using Telescope, and trying to save them with `:w`, I get the error `E13: File exists (add ! to override)`. After some trial and error, i found out that it was caused by an autocmd for the FileType event that was throwing an error. This doesn't happen when opening the files in other ways (like `:e`)
The main issue I think is that the error doesn't even get printed to `:mes`, so finding what the issue was was quite annoying. So if the `:w` error is not fixable, it'd be nice to at least get the error printed to help debugging

I found some open issues related to this, but they didn't seem to be able to reproduce the error consistently, so I'm opening a new one (with the minimal config provided I saw this behaviour 100% of the time). Previous issues:
- https://github.com/nvim-telescope/telescope.nvim/issues/2894
- https://github.com/nvim-telescope/telescope.nvim/issues/648
- https://github.com/nvim-telescope/telescope.nvim/issues/75

### Neovim version

```markdown
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1702233742

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"
```

### Operating system and version

Linux 6.7.6-arch1-1

### Telescope version / branch / rev

master

### checkhealth telescope

```markdown
telescope: require("telescope.health").check()

Checking for required plugins ~
- OK plenary installed.
- WARNING nvim-treesitter not found. (Required for `:Telescope treesitter`.)

Checking external dependencies ~
- OK rg: found ripgrep 14.1.0
- OK fd: found fd 9.0.0

===== Installed extensions ===== ~
```

### Steps to reproduce

1. Create some empty test file (say test.lua)
2. `nvim -nu minimal.lua`
3. `:Telescope find_files` and open test.lua
4. `:w`

### Expected behavior

I expect the error thrown in the FileType autocmd to be printed when opening the file, and I expect the file to be saved on `:w`

### Actual behavior

The error never gets printed, and when trying to save the file I get the error `E13: File exists (add ! to override)`

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          { 'nvim-telescope/telescope-fzf-native.nvim', run = 'make' },
        },
      },
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup()
  require('telescope').load_extension('fzf')

  vim.api.nvim_create_autocmd("FileType", {
    group = vim.api.nvim_create_augroup("my_aucmds", { clear = true }),
    callback = function()
      error "Some error"
    end,
  })
end

if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by jamestrew** [comment]
Thanks for the minimal config and debugging..
I think this is a neovim issue that has already been "fixed" upstream.
Trying this on nightly, I get an error immediately upon `:Telescope find_files` 
![image](https://github.com/nvim-telescope/telescope.nvim/assets/66286082/b2322162-678e-4dc1-9ea8-db04e607f727)

I'm still unable to `:w` the file, but at least the issue is more obvious now I hope.

**Comment by m-demare** [comment]
I just tried it on the latest neovim commit. Now it does print the errors, but only if they are thrown on the TelescopePrompt and TelescopeResults filetypes
If I modify the autocmd in the config above to only throw the error on some filetypes (say javascript), that error doesn't get printed and I still can't save with `:w`:
```lua
  vim.api.nvim_create_autocmd("FileType", {
    group = vim.api.nvim_create_augroup("my_aucmds", { clear = true }),
    callback = function(ev)
      print(ev.match)
      if ev.match == "javascript" then
        error "Some error"
      end
    end,
  })
```

**Comment by jamestrew** [comment]
Hm...
I'm still confident this isn't a Telescope issue. For example, I can replicate the original issue without Telescope with this config
```lua
local function prompt_fun()
  local bufnr = vim.api.nvim_create_buf(false, true)

  local win_opts = {
    relative = "editor",
    width = 10,
    height = 10,
    row = 10,
    col = 10,
    style = "minimal",
  }

  vim.api.nvim_open_win(bufnr, true, win_opts)
  vim.api.nvim_set_option_value("buftype", "prompt", { buf = bufnr })
  vim.api.nvim_set_option_value("ft", "MyPrompt", { buf = bufnr })

  vim.keymap.set({ "n", "i" }, "<esc>", function()
    vim.api.nvim_buf_delete(bufnr, { force = true })
    vim.cmd("e repro.lua")
  end, { buffer = bufnr })
end

vim.api.nvim_create_user_command("PromptFun", prompt_fun, {})

vim.api.nvim_create_autocmd("FileType", {
  group = vim.api.nvim_create_augroup("my_aucmds", { clear = true }),
  callback = function(ev)
    error("Some error")
  end,
})
```
Just run `:PromptFun` and hit `<esc>`.
You get a few more errors but the end result of not being able to save the buffer is present.
I can't replicate the issue when you hide the error behind a `ev.match == <ft>` but I hope you can appreciate that this is getting pretty deep into vim internals specific territory that I don't think Telescope will have much control over outside of sketchy workarounds.

I would normally consider opening an issue upstream, but I can see this very much being a low-priority/won't fix situation with high complexity without maybe a little bit more of a concrete idea of what's happening (or be able to replicate your second autocmd example with just regular neovim api).

**Comment by m-demare** [comment]
That's interesting, thanks for your help. I'll look into it and open an issue upstream if I find any more info

Btw I found an interesting behaviour change between nvim 0.9 and 0.10: in 0.9, the keymap is never set, because the autocmd is executed synchronously after setting the ft, which throws an error. To have the same behaviour as nvim 0.10, I had to move the vim.keymap.set before the set ft. Don't know if this was an intentional change, just thought it's interesting

Anyways, I'm closing this now since it doesn't seem to be a telescope issue
